tests/functional/syntax/modules/test_initializers.py

```diff
@@ -1208,3 +1208,23 @@ def test_ownership_decl_errors_not_swallowed(make_input_bundle):
     with pytest.raises(UndeclaredDefinition) as e:
         compile_code(main, input_bundle=input_bundle)
     assert e.value._message == "'lib2' has not been declared."
+
+
+def test_partial_compilation(make_input_bundle):
+    lib1 = """
+counter: uint256
+    """
+    main = """
+import lib1
+
+uses: lib1
+
+@internal
+def use_lib1():
+    lib1.counter += 1
+    """
+    input_bundle = make_input_bundle({"lib1.vy": lib1})
+    assert (
+        compile_code(main, input_bundle=input_bundle, output_formats=["annotated_ast_dict"])
+        is not None
```

Comment:
Very generic `assert` here IMO; can we assert it more specifically against the expected output?

Reply:
we actually don't need it since we are just calling `compile_code()` for its side effecs; it's just the convention in the whole test suite